src/core/pagerankGraph.test.js

```diff
@@ -165,6 +166,122 @@ describe("core/pagerankGraph", () => {
     });
   });
 
+  describe("edge filtering", () => {
+    const src1 = NodeAddress.fromParts(["src", "1"]);
+    const src2 = NodeAddress.fromParts(["src", "2"]);
+    const dst1 = NodeAddress.fromParts(["dst", "1"]);
+    const dst2 = NodeAddress.fromParts(["dst", "2"]);
+    const e11 = {
+      src: src1,
+      dst: dst1,
+      address: EdgeAddress.fromParts(["e", "1", "1"]),
+    };
+    const e12 = {
+      src: src1,
+      dst: dst2,
+      address: EdgeAddress.fromParts(["e", "1", "2"]),
+    };
+    const e21 = {
+      src: src2,
+      dst: dst1,
+      address: EdgeAddress.fromParts(["e", "2", "1"]),
+    };
+    const e22 = {
+      src: src2,
+      dst: dst2,
+      address: EdgeAddress.fromParts(["e", "2", "2"]),
+    };
+    const graph = () =>
+      [e11, e12, e21, e22].reduce(
+        (g, e) => g.addEdge(e),
+        [src1, src2, dst1, dst2].reduce((g, n) => g.addNode(n), new Graph())
+      );
+    const pagerankGraph = () => new PagerankGraph(graph(), defaultEvaluator);
+
+    function expectConsistentEdges(options: EdgesOptions | void) {
+      const pagerankGraphEdges = Array.from(pagerankGraph().edges(options));
+      pagerankGraphEdges.forEach((e) => {
+        expect(e.weight.froWeight).toBe(0);
+        expect(e.weight.toWeight).toBe(1);
+      });
+      const graphEdges = Array.from(graph().edges(options));
+      expect(pagerankGraphEdges.map((e) => e.edge)).toEqual(graphEdges);
+    }
+
+    describe("edge filter matches graph edge filter", () => {
+      it("finds all edges when no options are specified", () => {
```

Comment:
If you wanted to code golf this section, you could do so with something like:

```js
const mkOptions = (edgeParts, srcParts, dstParts) => ({
  addressPrefix: EdgeAddress.fromParts(edgeParts),
  srcPrefix: NodeAddress.fromParts(srcParts),
  dstPrefix: NodeAddress.fromParts(dstParts),
})

const testCases = [
  { description: "when no options are specified", options: null },
  {
    description: "when all-inclusive filters are specified", 
    options: mkOptions(["e"], ["src"], ["dst"])
  },
  ...
]

for (const case of testCases) {
  it(case.description,() => {
    expectConsistentEdges(case.options);
  })
}
```

Change not requested, since your current approach is fine + totally readable.
And we aren't running out of paper, so there's no _need_ for code golf. It can
be fun, though. :)

Reply:
My $0.02: I prefer the way that @BrianLitwin wrote it, primarily because
it’s simpler. Programmatically generating tests can be useful especially
when the parameters themselves are programmatically generated: e.g.,

<https://github.com/sourcecred/sourcecred/blob/656a2d15433f37f358027f03f7f0e54dd8b0fa01/src/plugins/github/render.test.js#L12-L19>

But here it’s not clear to me that you really buy anything from
generating the tests, and the cost is nonzero; for instance:

  - Reading `mkOptions(["e"], ["src"], ["dst"])` doesn’t tell you as
    much as reading the object with named fields. With the latter, not
    only are the parameters named so that you know which is which, but
    if you’re just jumping to this test case (which is likely, assuming
    that it’s just failed) then the names help you gain context on what
    kind of code this is testing.

  - The “golfed” form impedes searchability. If the test fails with an
    error message like “bad addressPrefix: `foo`”, I might grep the
    codebase or file for `addressPrefix.*foo`. With `mkOptions`, that
    won’t find the right thing. Similarly, `it.*all-inclusive filters`
    doesn’t find the test case definition.

(Definitely agreed that code golf can be fun, though, and you’ll often
learn a thing or two!)

Reply:
+1 to what William said. :)

Reply:
My $0.02 preference is try to keep the code in `graph.test` and `pagerankGraph.test` consistent because they're testing the same conditions.. what do y'all think about moving these graph/edge generators into a new test utility file (or `graphTestUtil`) and sharing them between `graph.test` and `pagerankGraph.test` ? 

Things that it would make easier: 
1) being certain the test cases are consistent and subsequent efforts to reason about test failures in one or both files. 
2) keeping the workflow/verbiage more consistent i.e. more readable.  E.g. "universal" vs "all-inclusive" filters are being used to describe the same edge filters in `graph.test` and `pagerankGraph.test`, respectively.  The same `graph` being created w/ `forEach` in one file and `reduce` in the other. 
2) adding new test cases 


And I think we could still golf 🏌️but keep the slightly more verbose version for clarity 
e.g.  => 
```javascript 
// create the edges, nodes, and graph in helper file 

const testCases = [
  {
    desc: "finds all edges when all-inclusive filters are specified",
    options: {
      addressPrefix: EdgeAddress.fromParts(["e"]),
      srcPrefix: NodeAddress.fromParts(["src"]),
      dstPrefix: NodeAddress.fromParts(["dst"]),
    }, 
    expected: [e11, e12, e21, e22]
  },
  {
    desc: "finds edges by address prefix",
    options: {
      addressPrefix: EdgeAddress.fromParts(["e", "1"]),
      srcPrefix: NodeAddress.empty,
      dstPrefix: NodeAddress.empty,
    },
   expected: [e11, e12]
  },
];

```

Reply:
I'm fine with a refactor to make the tests more consistent, but my preference would be that this commit merges more-or-less as is,  and then there's a separate pull that refactors the two to be consistent.

My reasoning being that there's some work I want to do that depends on having these methods in PagerankGraph, so preference towards merging it sooner. (Although admittedly there are a few other things I want to work on between now and then, so it's not a real rush.)